### PR TITLE
addition of the data-catalogue feature 

### PR DESCRIPTION
**Pull request by markhigham**

### CODE REVIEW

File: test/test.py [hunk 1]
```diff
@@ -272,7 +275,8 @@ def add_async_cleanup(self, coroutine):
             content = await response.text()
 
         self.assertIn(
-            '<a class="govuk-link" href="http://testapplication-23b40dd9.localapps.com:8000/" style="font-weight: normal;">Test Application</a>', content)
```
Comment (michalc) [inline test/test.py]: Test that `href="http://testapplication-23b40dd9.localapps.com:8000/"` is in the content if possible

File: app/app/templates/datagroup.html [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    <div class="govuk-breadcrumbs">
+        <ol class="govuk-breadcrumbs__list">
+            <li class="govuk-breadcrumbs__list-item">
+                <a class="govuk-breadcrumbs__link" href="/">Home</a>
+            </li>
+
+            <li class="govuk-breadcrumbs__list-item">
+                {{ model.name }}
+            </li>
+            {% if inline_edit and perms.app_datagrouping_change %}
```
Comment (michalc) [inline app/app/templates/datagroup.html]: No need for inline_edit: just check perms

File: app/app/templates/datagroup.html [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                <li class="govuk-breadcrumbs__list-item">
+                    <a target="_blank" class="govuk-breadcrumbs__link"
```
Comment (michalc) [inline app/app/templates/datagroup.html]: Consider removing _blank

File: app/app/templates/datagroup.html [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                </p>
+            {% endif %}
+        </div>
+    </div>
+
+
+    <div class="govuk-grid-row govuk-!-margin-top-8">
+        <div class="govuk-grid-column-two-thirds">
+            <h2 class="govuk-heading-l">Additional Information</h2>
+            <table class="govuk-table catalogue-table">
```
Comment (michalc) [inline app/app/templates/datagroup.html]: Consider summary list

File: app/app/templates/root.html [hunk 5]
```diff
@@ -10,53 +10,138 @@
...
+                                <dt class="govuk-summary-list__key">
+                                    <a class="govuk-link" href="{{ appstream_url }}"
+                                       style="font-weight: normal;">AppStream</a>
+                                </dt>
+                                <dd class="govuk-summary-list__actions">
+                                </dd>
+                            </div>
+                        {% endif %}
+                    </dl>
+                </div>
```
Comment (michalc) [inline app/app/templates/root.html]:
```
if groupings:
   sdfsdf


if schemas:
  sdfsdf


if not groupings and not schemas:
  No data is availabled


--------


if has app permisssion:
  .....
```

File: app/app/views_catalogue.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        'model': item,
+        'datasets': item.dataset_set.all().order_by('name')
+    }
+
+    return render(request, 'datagroup.html', context)
+
+
+@require_GET
+def dataset_full_path_view(request, group_slug, set_slug):
+    found = DataSet.objects.filter(grouping__slug=group_slug, slug=set_slug)[0]
```
Comment (michalc) [inline app/app/views_catalogue.py]: Minor thing, but will this result in a 500 if there is not matching group or set slug?

File: app/app/views.py [hunk 7]
```diff
@@ -102,7 +106,7 @@ def can_stop(application_template):
             for database, database_privilages in privilages_by_database
         ])),
         'appstream_url': settings.APPSTREAM_URL,
-        'support_url': settings.SUPPORT_URL,
```
Comment (markhigham) [inline app/app/views.py]: this is defined in `context_processors.py`?

Comment (michalc) [reply app/app/views.py]: Yes: it didn't use to be, and I then refactored it out (but looks like I missed a place where it was originally defined)

File: app/app/templates/root.html [hunk 8]
```diff
@@ -10,53 +10,78 @@
...
+            <h1 class="govuk-heading-l">Data</h1>
+            {% if database_schema_tables %}
+                <ul class="govuk-list">
+                    {% for database_schema_table in database_schema_tables %}
+                        <li><a class="govuk-link"
+                               href="{% url 'table_data' database_schema_table.0 database_schema_table.1 database_schema_table.2 %}">{{ database_schema_table.0 }}
+                            / {{ database_schema_table.1 }} / {{ database_schema_table.2 }}</a></li>
+                    {% endfor %}
+                </ul>
+            {% else %}
```
Comment (michalc) [inline app/app/templates/root.html]: Should the no data is available only be shown if `not database_schema_tables and not groupings` ?

**Pull request merged**